dcc-portal-ui/app/scripts/pathwayviewer/js/controller.js

```diff
@@ -232,30 +238,45 @@ angular.module('icgc.pathwayviewer.directives.controller', ['icgc.pathwayviewer.
        *  and transforms and renders with the form:
        *  [{id:123, value:10},{id:124, value:10},...]
        */
-      ReactomePathway.prototype.highlight = function (rawHighlights, overlaps) {
-        var highlights = [];
-        var nodesInPathway = this.nodesInPathway;
-        console.log(nodesInPathway);
-        rawHighlights.forEach(function (rh) {
-          rh.dbIds.forEach(function (dbId) {
-
-            // Only highlight it if it's part of the pathway we're zooming in on
-            // And only hide parts of it we are zooming in on a pathway
-            if((nodesInPathway.length === 0 || _.contains(nodesInPathway,dbId)) && rh.value >= 0){
-              highlights.push({id:dbId,value:rh.value});
-            }
-          });
-        });
-
+      ReactomePathway.prototype.highlight = function (rawMutationHighlights, rawDrugHighlights, overlaps) {
         _.keys(overlaps).forEach(function (dbId) {
             // Only highlight overlaps it if it's part of the pathway we're zooming in on
             // And only hide parts of it we are zooming in on a pathway
-            if((nodesInPathway.length !== 0 && ! _.contains(nodesInPathway,dbId))){
+            if((this.nodesInPathway.length !== 0 && ! _.contains(this.nodesInPathway,dbId))){
               delete overlaps[dbId];
             }
         });
 
-        this.renderer.highlightEntity(highlights, overlaps, this.model);
+        this.renderer.highlightEntity(
+            _retrieveHighlights('mutation', rawMutationHighlights, this.nodesInPathway), 
+            _retrieveHighlights('drug', rawDrugHighlights, this.nodesInPathway),
+            overlaps,
+            this.model
+        );
+        
+        function _retrieveHighlights(type, rawHighlights, nodesInPathway) {
+          var highlights = [];
+          rawHighlights.forEach(function (rawHighlight) {
+            rawHighlight.dbIds.forEach(function (dbId) {
+              // Only highlight it if it's part of the pathway we're zooming in on
+              // And only hide parts of it we are zooming in on a pathway
+              if((nodesInPathway.length === 0 || _.contains(nodesInPathway,dbId)) && 
+                  _getHighlightValue(rawHighlight, type) >= 0){
+                highlights.push({id:dbId,value:_getHighlightValue(rawHighlight, type)});
+              }
+            });
+          });
+          return highlights;
+        }
+        
```

Comment:
This one might be tricky, since he the last map needs access to rawHighlight 

The version of lodash we're using also does not seem to support flatMap :(
We're also using a lot of methods that have been removed in the current version of lodash so we'll have to lint through our code before we can safely upgrade

Should be able to manually map and flatten for this function though

```
_(rawHighlights)
            .map(function (rawHighlight) {
              var highlightValue = _getHighlightValue(rawHighlight, type);
              return _(rawHighlight.dbIds)
                .filter(function (dbId) {
                  return (nodesInPathway.length === 0 || _.contains(nodesInPathway, dbId)) && highlightValue >= 0;
                })
                .map(function (dbId) {
                  return {id: dbId, value: highlightValue };
                })
                .value();
            })
            .flatten()
            .value();
```

In this specific case though I'm not sure if this is better

Although it does look like the value for `_getHighlightValue(rawHighlight, type)` should be stored in a variable and moved outside the inner forEach